server/src/com/thoughtworks/go/config/update/UpdateSCMConfigCommand.java

```diff
@@ -18,33 +18,62 @@
 
 
 import com.thoughtworks.go.config.CruiseConfig;
-import com.thoughtworks.go.config.commands.EntityConfigUpdateCommand;
 import com.thoughtworks.go.domain.scm.SCM;
 import com.thoughtworks.go.domain.scm.SCMs;
 import com.thoughtworks.go.i18n.LocalizedMessage;
 import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.EntityHashingService;
 import com.thoughtworks.go.server.service.GoConfigService;
 import com.thoughtworks.go.server.service.materials.PluggableScmService;
 import com.thoughtworks.go.server.service.result.LocalizedOperationResult;
 import com.thoughtworks.go.serverhealth.HealthStateType;
 
 public class UpdateSCMConfigCommand extends SCMConfigCommand {
 
-    public UpdateSCMConfigCommand(SCM globalScmConfig, PluggableScmService pluggableScmService, GoConfigService goConfigService, Username currentUser, LocalizedOperationResult result) {
+    private String md5;
+    private EntityHashingService entityHashingService;
+
+    public UpdateSCMConfigCommand(SCM globalScmConfig, PluggableScmService pluggableScmService, GoConfigService goConfigService, Username currentUser, LocalizedOperationResult result, String md5, EntityHashingService entityHashingService) {
         super(globalScmConfig, pluggableScmService, goConfigService, currentUser, result);
+        this.md5 = md5;
+        this.entityHashingService = entityHashingService;
     }
 
     @Override
     public void update(CruiseConfig modifiedConfig) throws Exception {
-        SCMs scms = modifiedConfig.getSCMs();
-        SCM scm = scms.find(globalScmConfig.getSCMId());
-        if (scm != null) {
-            scm.setAutoUpdate(globalScmConfig.isAutoUpdate());
-            scm.setConfiguration(globalScmConfig.getConfiguration());
+        SCM scm = findSCM(modifiedConfig);
+        scm.setAutoUpdate(globalScmConfig.isAutoUpdate());
+        scm.setConfiguration(globalScmConfig.getConfiguration());
+
+    }
+
+
+    @Override
+    public boolean canContinue(CruiseConfig cruiseConfig) {
+        return isUserAuthorized() && isRequestFresh(cruiseConfig);
+    }
+
+    private boolean isRequestFresh(CruiseConfig cruiseConfig) {
+        SCM existingSCM = findSCM(cruiseConfig);
+        return entityHashingService.md5ForEntity(existingSCM, existingSCM.getName()).equals(md5);
+    }
```

Comment:
can we add a message to the result if it is stale request
something like ` result.stale()`